Wikipedia/Code/ActivityTabFunnel.swift

```diff
@@ -0,0 +1,124 @@
+            case platform = "platform"
+            case wikiID = "wiki_id"
+        }
+    }
+    
+    private func logEvent(activeInterface: ActiveInterface? = nil, action: Action? = nil, actionData: [String: String]? = nil, project: WikimediaProject? = nil) {
+        var actionDataString: String? = nil
+        if let actionData {
+            actionDataString = ""
+            for (key, value) in actionData {
+                actionDataString?.append("\(key):\(value), ")
+            }
+            
+            if let finalActionDataString = actionDataString,
+               finalActionDataString.count > 1 {
+                actionDataString?.removeLast(2)
+            }
+        }
+        
+        let event = ActivityTabFunnel.Event(activeInterface: activeInterface?.rawValue, action: action?.rawValue, actionData: actionDataString, platform: "ios", wikiID: project?.notificationsApiWikiIdentifier)
+        EventPlatformClient.shared.submit(stream: .appActivityTab, event: event)
+    }
+    
+    func logOnboardingDidAppear() {
+        logEvent(activeInterface: .activityTabStart, action: .impression, project: nil)
+    }
+    
+    func logOnboardingDidTapContinue() {
+        logEvent(activeInterface: .activityTabStart, action: .continueClick, project: nil)
+    }
+    
+    func logOnboardingDidTapLearnMore() {
+        logEvent(activeInterface: .activityTabStart, action: .learnClick, project: nil)
+    }
+    
+    func logGroupAssignment(group: String) {
+        logEvent(activeInterface: nil, action: .launch, actionData: ["group": group], project: nil)
+    }
+    
+    func logActivityTabImpressionState(empty: String) {
+        logEvent(activeInterface: .activityTab, action: .impression, actionData: ["state": empty], project: nil)
+    }
+    
+    func logActivityTabArticleTap() {
+        logEvent(activeInterface: .activityTab, action: .articleClick, project: nil)
+    }
+    
+    func logActivityTabOverflowMenuLearnMore() {
+        logEvent(activeInterface: .overflowMenu, action: .learnClick, project: nil)
+    }
+    
+    func logActivityTabOverflowMenuClearHistory() {
+        logEvent(activeInterface: .overflowMenu, action: .clearHistory, project: nil)
+    }
+    
+    func logActivityTabOverflowMenuProblem() {
+        logEvent(activeInterface: .overflowMenu, action: .problemClick, project: nil)
+    }
+    
+    func logActivityTabSurveyImpression() {
+        logEvent(activeInterface: .survey, action: .impression, project: nil)
+    }
+    
+    func logActivityTabSurveyCancel() {
+        logEvent(activeInterface: .survey, action: .surveyClose, project: nil)
+    }
+    
+    func logFeedbackSubmit(selectedItems: [String], comment: String?) {
+        let selectedJoined = selectedItems.filter { $0 != "other" }.joined(separator: ",")
+        var actionData = [
+            "feedback_select": selectedJoined
+        ]
+
+        if let comment {
+            actionData["feedback_comment"] = comment
```

Comment:
I think when users submit comments with commas, it messes up the backend somehow. For year in review I replaced them [here](https://github.com/wikimedia/wikipedia-ios/blob/230a2c7d5175ae8d43dfe75e639f7d76ad420c77/Wikipedia/Code/DonateFunnel.swift#L520) before submitting. I will check to see if that same approach will work here.